File: c#/funnyAddition.cs
using System;

namespace ConsoleApp8
{
    class Program
    {
        static string reverse (string n)
        {
            char[] CharArray = n.ToCharArray();
            Array.Reverse(CharArray);
            return new String(CharArray);
        }
        static string palindrome(int n)
        {
            int count = 0;
            while (n != Convert.ToInt32(reverse(n.ToString())))
            {
                n = n + Convert.ToInt32(reverse(n.ToString()));
                count += 1;
            }
            return n.ToString() + " " + count.ToString();
            
        }
        static void Main(string[] args)
        {
            Console.WriteLine("Podaj liczbę: ");
            string n = Console.ReadLine();
            Console.WriteLine(reverse(n));
            Console.WriteLine(palindrome(Convert.ToInt32(n)));
        }
    }
}

File: FunnyAddition.java
package com.company;

public class FunnyAddition {
    public String reverse(int n){
        return new StringBuilder(String.valueOf(n)).reverse().toString();
    }

    public String palindrome(int number) {
        int count = 0;
        while (number != Integer.valueOf(reverse(number))){
            number = number + Integer.valueOf(reverse(number));
            count += 1;
        }
        System.out.print(String.valueOf(number));
        return String.valueOf(number);
    }
}
